### PR TITLE
Rest api config

### PR DESCRIPTION
**Pull request by kelmag**
in this branch, I added a fake REST API to manipulate data inside the project using javascript's promises to fetch and store the data in the redux store
NB : still I haven't worked on the user login, so for the moment, it's just the posts part that is fetched from the API

### CODE REVIEW

File: src/pages/Profile/index.js [hunk 1]
```diff
@@ -1,74 +1,87 @@
...
-export default Login;
+const enhance = compose(
+  lifecycle({
+    componentDidMount() {
+      const {
+        fetchPostRequest,
+        fetchPostRequestFailure,
+        fetchPostRequestSuccess
+      } = this.props;
+      fetchPostRequest();
```
Comment (abumalick) [inline src/pages/Profile/index.js]: we can use redux thunk for async actions. https://github.com/reduxjs/redux-thunk

It is the simplest way in my opinion.

File: src/pages/Profile/index.js [hunk 2]
```diff
@@ -1,74 +1,87 @@
...
+  items: state.items,
+  loading: state.loading,
+  error: state.error
+});
+
+const mapDispatchToProps = dispatch => ({
+  fetchPostRequest: () => dispatch(fetchPostRequest()),
+  fetchPostRequestSuccess: items => dispatch(fetchPostRequestSuccess(items)),
+  fetchPostRequestFailure: error => dispatch(fetchPostRequestFailure(error))
+});
```
Comment (abumalick) [inline src/pages/Profile/index.js]: there is a simpler way to do this if you want:
```
const mapDispatchToProps =  {
  fetchPostRequest,
  fetchPostRequestSuccess,
  fetchPostRequestFailure
};
```

File: src/pages/Profile/index.js (hunk 1 above)
Comment (kelmag) [reply src/pages/Profile/index.js]: I'll work on it on the next commit

File: src/models/posts/actions.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                    })
+                    .then(posts => {
+                      //returning an array of promises each one contains a post being parsed
+                      let promises = posts.map(post => {
+                            return postParser(post)
+                              .then(parsedPost => parsedPost)
+                              .catch(err => console.log(err));
+                          });
+                      // return the promises 
+                      return Promise.all(promises).then(results => results);
```
Comment (abumalick) [inline src/models/posts/actions.js]: the then is not necessary here

File: src/utils/myDataParser.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import DB from "db.js";
+
+export { DB };
+
+export const getPost = postId => {
+  return DB.posts.filter(post => post.id === postId);
+};
+
+export const getUser = id => {
+  return new Promise((resolve, reject) => {
```
Comment (abumalick) [inline src/utils/myDataParser.js]: you can return fetch directly here I think

fetch returns a promise
```
export const getUser = id => {
  return fetch(`https://my-json-server.typicode.com/kelmag/demo/users/${id}`)
      .then(response => {
        if (response.status >= 400) throw new Error("Bad response from server");
        return response.json();
      })
      .catch(err => console.log(err));
};

**Comment by kelmag** [comment]
> good work implementing redux thunk, I didn't know about redux-promise-middleware, looks good.
> 
> It looks cleaner, by removing the fetching logic from the code 👍
> 
> Good work !

thank you!

**Comment by kelmag** [comment]
I have added a local json server in the app and then added the redux action CREATE_POST_REQUEST to create a new post in the database and then update the state in the redux store

**Pull request merged**